pkg/apiserver/resthandler.go

```diff
@@ -446,6 +417,93 @@ func PatchResource(r rest.Patcher, scope RequestScope, typer runtime.ObjectTyper
 
 		write(http.StatusOK, scope.APIVersion, scope.Codec, result, w, req.Request)
 	}
+
+}
+
+// patchResource divides PatchResource for easier unit testing
+func patchResource(ctx api.Context, timeout time.Duration, versionedObj runtime.Object, r rest.Patcher, name string, patchType api.PatchType, patchJS []byte, namer ScopeNamer, codec runtime.Codec) (runtime.Object, error) {
+	namespace := api.NamespaceValue(ctx)
+
+	original, err := r.Get(ctx, name)
+	if err != nil {
+		return nil, err
+	}
+
+	originalObjJS, err := codec.Encode(original)
+	if err != nil {
+		return nil, err
+	}
+	originalPatchedObjJS, err := getPatchedJS(patchType, originalObjJS, patchJS, versionedObj)
+	if err != nil {
+		return nil, err
+	}
+
+	objToUpdate := r.New()
+	if err := codec.DecodeInto(originalPatchedObjJS, objToUpdate); err != nil {
+		return nil, err
+	}
+	if err := checkName(objToUpdate, name, namespace, namer); err != nil {
+		return nil, err
+	}
+
+	return finishRequest(timeout, func() (runtime.Object, error) {
+		// update should never create as previous get would fail
+		updateObject, _, updateErr := r.Update(ctx, objToUpdate)
+		if errors.IsConflict(updateErr) {
```

Comment:
What if there is a second conflict? I think this would make more sense as a loop.